service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -225,20 +229,25 @@ public function passportNumberAction(): ViewModel
             $view->setVariable('passport', $data['passport']);
 
             if (array_key_exists('check_button', $formData->toArray())) {
-                $formProcessorResponseDto = $this->formProcessorHellper->processPassportDateForm(
+                $formProcessorResponseDto = $this->formProcessorHelper->processPassportDateForm(
                     $uuid,
                     $this->getRequest()->getPost(),
                     $dateSubForm,
                     $templates
                 );
             } else {
-                $view->setVariable('passport_indate', ucwords($data['inDate']));
-                $formProcessorResponseDto = $this->formProcessorHellper->processPassportForm(
+                $formProcessorResponseDto = $this->formProcessorHelper->processPassportForm(
                     $uuid,
                     $this->getRequest()->getPost(),
                     $form,
                     $templates
                 );
+                $view->setVariable(
+                    'passport_indate',
+                    array_key_exists('inDate', $data) ?
+                        ucwords($data['inDate']) :
+                        'no'
+                );
```

Comment:
This would be much neater if `inDate` was a boolean, then the template could take care of translation rather than doing it inside the data model.